hazelcast/src/main/java/com/hazelcast/instance/impl/DefaultNodeExtension.java

```diff
@@ -297,6 +297,24 @@ public SecurityContext getSecurityContext() {
 
     @Override
     public InternalSerializationService createSerializationService() {
+        return createSerializationService(false);
+    }
+
+    @Override
+    public InternalSerializationService createCompatibilitySerializationService() {
```

Comment:
since we only do this change for version 3 compatibility,  maybe we can be explicit in naming in this PR and use something like:
`createCompatibilitySerializationService `--> `createSerializationServiceForHzV3` 
`isCompatibility`--> `isForHzV3`